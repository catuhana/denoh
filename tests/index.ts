import { assertSnapshot } from 'https://deno.land/std@0.177.0/testing/snapshot.ts';

Deno.test('createGitHookScript() tests', async (t) => {
  const createGitHookScript =
    (await import('../src/index.ts')).createGitHookScript;

  await assertSnapshot(t, createGitHookScript(['lint']));
  await assertSnapshot(
    t,
    createGitHookScript([
      '!echo \'Changed branch, running lint tasks...\'',
      'lint ; lint:fmt',
      '!echo \'Tasks ran successfully.\'',
    ]),
  );
  await assertSnapshot(t, createGitHookScript(['lint ; fmt && test ; notify']));
  await assertSnapshot(t, createGitHookScript(['test1 && test2 && test3']));
});

Deno.test('getGitHooks() tests', async (t) => {
  const getGitHooks = (await import('../src/index.ts')).getGitHooks;

  assertSnapshot(t, (await getGitHooks('tests/deno.test.jsonc')).githooks);
});

Deno.test('createHooks() tests', async (t) => {
  const createHooks = (await import('../src/index.ts')).createHooks;

  assertSnapshot(t, (await createHooks('tests/deno.test.jsonc')).createdHooks);
});
