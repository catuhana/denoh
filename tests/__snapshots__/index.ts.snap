export const snapshot = {};

snapshot[`createGitHookScript() tests 1`] = `
"#!/bin/sh
deno task lint"
`;

snapshot[`createGitHookScript() tests 2`] = `
"#!/bin/sh
echo 'Changed branch, running lint tasks...'
deno task lint ; deno task lint:fmt
echo 'Tasks ran successfully.'"
`;

snapshot[`createGitHookScript() tests 3`] = `
"#!/bin/sh
deno task lint ; deno task fmt && deno task test ; deno task notify"
`;

snapshot[`createGitHookScript() tests 4`] = `
"#!/bin/sh
deno task test1 && deno task test2 && deno task test3"
`;

snapshot[`getGitHooks() tests 1`] = `
{
  configPath: "C:\\\\Users\\\\Tuhana\\\\Documents\\\\Projects\\\\denoh\\\\tests",
  githooks: {
    "post-checkout": [
      "!echo 'Changed branch, running lint tasks...'",
      "lint ; lint:fmt",
      "!echo 'Tasks ran successfully.'",
    ],
    "post-commit": [
      "!echo 'Added commit'",
    ],
    "pre-commit": [
      "lint",
    ],
  },
}
`;

snapshot[`createHooks() tests 1`] = `
{
  createdHooks: [
    {
      gitHookName: "pre-commit",
      gitHookScript: "#!/bin/sh
deno task lint",
    },
    {
      gitHookName: "post-commit",
      gitHookScript: "#!/bin/sh
echo 'Added commit'",
    },
    {
      gitHookName: "post-checkout",
      gitHookScript: "#!/bin/sh
echo 'Changed branch, running lint tasks...'
deno task lint ; deno task lint:fmt
echo 'Tasks ran successfully.'",
    },
  ],
  hooksPath: "C:\\\\Users\\\\Tuhana\\\\Documents\\\\Projects\\\\denoh\\\\tests",
}
`;
